docs/source/azure/policy/genericarmaction.rst

```diff
@@ -121,7 +121,7 @@ Others
               - name: delete-test-resources
                 description: |
                   Deletes any ARM resource with 'test' in the name
-                resource: azure.armresource
+                resource: azure.arm-resource
```

Comment:
this feels like it will introduce an inconsistency, every other azure resource with compound words in the name does lower case without hypens.. sqlserver, networksecuritygroup, appserviceplan, keyvault.

Reply:
1. Yes, same as the ones for graph queries
2. This is a question I wanted to discuss.. I think other places require hyphens too. 
I see that gcp uses hyphens, AWS has a mix of approaches.
I would suggest add hyphens for other Azure resources as an alias & add code to log warnings if old names are in use with a message like 'please update, this alias will be deprecated in X months' and drop old names at some point in the future. What do you think?

Reply:
i think its more important that a provider resource naming is consistent with itself then across all providers. aws has histerical raisons, we've evolved to be more civilized ;-) 

wrt to aliases, we could, but i'd prefer to avoid resource aliases atm (at least for till q3), we do support it for actions/filters, but there's alot of places on schema gen / schema cli / and docs where we have to dedup, and thats currently not shared logic. t

getting a proper deprecation framework is still on the todo list, its a blocker for 1.0. there's a separate issue for it. we'd have to keep to semver. technically 1.0 gives us a chance to break but we would need to be at 0.9 for at least six months prior to that.

Reply:
We want to evolve too! 😄 
Reverted to `armresource` for now